sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1625,6 +1625,14 @@ object SQLConf {
     .intConf
     .createWithDefault(25)
 
+  val MAX_PLAN_STRING_LENGTH = buildConf("spark.sql.maxPlanLength")
+    .doc("Maximum number of characters to output for a plan string.  If the plan is " +
+      "longer, further output will be truncated.  The default setting always generates a full " +
+      "plan.  Set this to a lower value such as 8192 if plan strings are taking up too much " +
+      "memory or are causing OutOfMemory errors in the driver or UI processes.")
+    .intConf
```

Comment:
Can you check that the value set is > 0? You can use the `checkValue(_ > 0, "...")` function for this.

Also use `ByteArrayMethods.MAX_ROUNDED_ARRAY_LENGTH` instead of `Int.MaxValue` as the default here.